hail/src/main/scala/is/hail/cxx/Emit.scala

```diff
@@ -797,6 +797,59 @@ class Emitter(fb: FunctionBuilder, nSpecialArgs: Int, ctx: SparkFunctionContext)
            """.stripMargin,
           resultRegion)
 
+      case ir.MakeNDArray(data, shape, rowMajor) =>
+        val elementPType = data.pType.asInstanceOf[PContainer].elementType
+        val datat = emit(data)
+        val shapet = emit(shape)
+        val rowMajort = emit(rowMajor)
+
+        val elemSize = elementPType.byteSize
+        val elemAlignment = elementPType.alignment
+        val flags = fb.variable("flags", "int", s"${rowMajort.v} ? 1 : 0")
+        val shapeVec = fb.variable("shapeVec", "std::vector<long>",
+          s"load_vector<long, true, 8, 8>(${shapet.v})")
+        val entries = fb.variable("entries", "const char *",
+          s"ArrayAddrImpl<true, $elemSize, $elemAlignment>::load_element(${datat.v}, 0)")
+        triplet(
```

Comment:
there's a shorthand for non-missing triplets---`present(v)`---that you could use here. It involves pushing the setup into the value computation, which seems fine.

Reply:
Ah nice I'll do that.